test/server/charset-test.js

```diff
@@ -0,0 +1,329 @@
+var fs                  = require('fs');
+var request             = require('request');
+var expect              = require('chai').expect;
+var express             = require('express');
+var iconv               = require('iconv-lite');
+var Proxy               = require('../../lib/proxy');
+var Session             = require('../../lib/session');
+var requestAgent        = require('../../lib/pipeline/request-agent');
+var Charset             = require('../../lib/pipeline/charset');
+var contentUtils        = require('../../lib/utils/content');
+var urlUtil             = require('../../lib/utils/url');
+var scriptProcessor     = require('../../lib/processing/resources/script');
+var pageProcessor       = require('../../lib/processing/resources/page');
+var stylesheetProcessor = require('../../lib/processing/resources/stylesheet');
+var manifestProcessor   = require('../../lib/processing/resources/manifest');
+
+function normalizeCode (code) {
+    return code
+        .replace(/(\r\n|\n|\r)/gm, ' ')
+        .replace(/'/gm, '"')
+        .replace(/\s+/gm, ' ');
+}
+
+function compareCode (code1, code2) {
+    expect(normalizeCode(code1)).eql(normalizeCode(code2));
+}
+
+describe('Content charset', function () {
+    var destServer = null;
+    var proxy      = null;
+    var session    = null;
+
+    var proxyHostname = '127.0.0.1';
+    var proxyPort1    = 1836;
+    var proxyPort2    = 1837;
+    var sessionId     = 'sessionId';
+
+    var pageWithMetaSrc    = fs.readFileSync('test/server/data/content-charset/page-with-meta.htm').toString();
+    var pageWithoutMetaSrc = fs.readFileSync('test/server/data/content-charset/page-without-meta.htm').toString();
+    var scriptSrc          = fs.readFileSync('test/server/data/content-charset/script.js').toString();
+    var manifestSrc        = fs.readFileSync('test/server/data/content-charset/manifest').toString();
+    var stylesheetSrc      = fs.readFileSync('test/server/data/content-charset/style.css').toString();
+
+    var charsetBOM         = 'utf-16be';
+    var charsetContentType = 'utf-8';
+    var charsetMeta        = 'windows-1251';
+    var charsetDefault     = 'iso-8859-1';
+    var charsetUrl         = 'utf-16le';
```

Comment:
Same here, just pass them directly, moreover their naming is not semantically correct.